### PR TITLE
support crossbuild with container

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:
support crossbuild with container
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
Whether we can add another ci job for multiple build in container, build in host maybe still need to be tested?

**Comment by gy95** [comment]
> Whether we can add another ci job for multiple build in container, build in host maybe still need to be tested?

seems no need anymore, building inside container is almost the same as building with the local environment.

**Comment by fisherxu** [comment]
> > Whether we can add another ci job for multiple build in container, build in host maybe still need to be tested?
> 
> seems no need anymore, building inside container is almost the same as building with the local environment.

How do you think? @zc2638 @zhu733756

File: hack/make-rules/build_with_container.sh [hunk 1]
```diff
@@ -25,11 +25,10 @@ MOUNTPATH="${MOUNTPATH:-/kubeedge}"
 KUBEEDGE_BUILD_IMAGE=${KUBEEDGE_BUILD_IMAGE:-"kubeedge/build-tools"}
 DOCKER_GID="${DOCKER_GID:-$(grep '^docker:' /etc/group | cut -f3 -d:)}"
 
-set -x
+echo "building inside container"
```
Comment (fisherxu) [inline hack/make-rules/build_with_container.sh:28]:
```suggestion
echo "start building inside container"
```
add a start?

Comment (gy95) [reply hack/make-rules/build_with_container.sh]: yes

**Comment by zhu733756** [comment]
LGTM. Can we make the image reused in CI by uploading the artifacts like https://github.com/actions/upload-artifact?

**Comment by fisherxu** [comment]
> LGTM. BTW, can we make the image reused in CI by uploading the artifacts like https://github.com/actions/upload-artifact?

Sure, can download the image once,  used in all jobs for one PR?

**Comment by zhu733756** [comment]
> > LGTM. BTW, can we make the image reused in CI by uploading the artifacts like https://github.com/actions/upload-artifact?
> 
> Sure, can download the image once, used in all jobs for one PR?

Yes, the other CI would need the build process.

**Comment by fisherxu** [comment]
Ref: https://stackoverflow.com/a/65369271/12970573

**Comment by fisherxu** [comment]
We can downlaod the build-tools image now? Or other images also needed?

**Review by fisherxu — COMMENTED** [review]
/approve
cc @zhu733756 PTAL, btw, can we split the one `main.yaml` to several files for every job?

**Comment by zhu733756** [comment]
I wonder whether we want to refactor the scripts like keadm_e2e_tesh.sh. If we can reuse cloudcore or build-tools or other images in a CI is awesome.

**Comment by fisherxu** [comment]
> I wonder whether we want to refactor the scripts like keadm_e2e_test.sh. If we can reuse cloudcore or build-tools or other images in a CI is awesome.

Sounds good.
/lgtm

**Pull request merged**